service-chain-aws/deploy-multi_layer-servicechain/common.tf

```diff
@@ -0,0 +1,160 @@
+  type                     = "ingress"
+  from_port                = 50505
+  to_port                  = 50506
+  protocol                 = "tcp"
+  cidr_blocks              = ["${aws_eip.sen[count.index].public_ip}/32"]
+}
+
+resource "aws_security_group" "l2_common_sg" {
+  name        = "${var.security_group}-layer2-sg"
+  description = "ServiceChain Layer 2 security group"
+  vpc_id      = var.vpc_id
+  tags        = var.tags
+}
+
+resource "aws_security_group_rule" "ingress_l2_ssh_tcp" {
+  security_group_id = aws_security_group.l2_common_sg.id
+  type              = "ingress"
+  from_port         = 22
+  to_port           = 22
+  protocol          = "tcp"
+  cidr_blocks       = var.ssh_client_ips
+}
+
+resource "aws_security_group_rule" "ingress_l2_network_tcp" {
+  security_group_id        = aws_security_group.l2_common_sg.id
+  type                     = "ingress"
+  from_port                = 32323
+  to_port                  = 32324
+  protocol                 = "tcp"
+  source_security_group_id = aws_security_group.l2_common_sg.id
+}
+
+resource "aws_security_group_rule" "ingress_l2_network_udp" {
+  security_group_id        = aws_security_group.l2_common_sg.id
+  type                     = "ingress"
+  from_port                = 32323
+  to_port                  = 32323
+  protocol                 = "udp"
+  source_security_group_id = aws_security_group.l2_common_sg.id
+}
+
+resource "aws_security_group_rule" "ingress_l2_scn_eip_network_tcp" {
+  count                    = var.scn_public_ip ? var.scn_instance_count : 0
+  security_group_id        = aws_security_group.l2_common_sg.id
+  type                     = "ingress"
+  from_port                = 32323
+  to_port                  = 32324
+  protocol                 = "tcp"
+  cidr_blocks              = ["${aws_eip.scn[count.index].public_ip}/32"]
+}
+
+resource "aws_security_group_rule" "ingress_l2_scn_eip_network_udp" {
+  count                    = var.scn_public_ip ? var.scn_instance_count : 0
+  security_group_id        = aws_security_group.l2_common_sg.id
+  type                     = "ingress"
+  from_port                = 32323
+  to_port                  = 32323
+  protocol                 = "udp"
+  cidr_blocks              = ["${aws_eip.scn[count.index].public_ip}/32"]
+}
+
+resource "aws_security_group_rule" "ingress_l2_en_eip_network_tcp" {
+  security_group_id        = aws_security_group.l2_common_sg.id
+  type                     = "ingress"
+  from_port                = 32323
+  to_port                  = 32324
+  protocol                 = "tcp"
+  source_security_group_id = var.en_security_group_id
+}
+
+resource "aws_security_group_rule" "ingress_l2_en_eip_network_udp" {
+  security_group_id        = aws_security_group.l2_common_sg.id
+  type                     = "ingress"
+  from_port                = 32323
+  to_port                  = 32323
+  protocol                 = "udp"
+  source_security_group_id = var.en_security_group_id
+}
```

Comment:
Don't we need to open port 50505 and 50506 for `en_security_group_id`?

Reply:
On a second thought, not necessarily. I've heard that sub-bridge acts as client and connects to main-bridge port, so allowing inbound connection to main-bridge port from sub-bridge port would be sufficient. However, I added rule for inbound connection to sub-bridge port from main-bridge to sync with existing codes and reduce confusion.

Reply:
Resolved in 4f1be1c